mlir/lib/ExecutionEngine/conv-validation-wrappers.cpp

```diff
@@ -158,8 +158,16 @@ void mcpuVerify(T *gpuResults, T *validationResults, long long dataSize,
       if ((gpuNum == 0.0f || (std::fpclassify(gpuNum) == FP_SUBNORMAL)) &&
           (std::signbit(valNum) == std::signbit(gpuNum))) {
         hist_relDiff[0]++;
+      } else if (std::is_same<T, float>::value) {
```

Comment:
You probably want to check this first and then do sign bit comparison in later else if.